violet-framework/src/main/java/com/horstmann/violet/workspace/editorpart/behavior/CutCopyPasteBehavior.java

```diff
@@ -195,16 +193,76 @@ public void paste()
 
             editorPart.getSwingComponent().invalidate();
             editorPart.getSwingComponent().repaint();
+
+            validatePaste(nodes, graph);
         }
         catch (IOException e)
         {
-            // Nothing to do
         }
     }
 
+    private void validatePaste(List<INode> nodes, IGraph graph){
+        if(checkIfNumberOfInitialNodesIsMoreThanOne(nodes, graph))
+        {
+            showWarningAboutNumberOfInitialNodes();
+        }
+    }
+    /**
+     * Is checking for each copied node if there is too many initialNodes on workspace
+     *
+     * @param nodes
+     * @param graph
+     * @return true if number of initialNodes is more than one
+     */
+    private boolean checkIfNumberOfInitialNodesIsMoreThanOne(List<INode> nodes, IGraph graph)
+    {
+        Collection<INode> nodeCollection = graph.getAllNodes();
+        for (INode newNode : nodes)
+        {
+            if (isNumberOfInitialNodesIsToHigh(newNode, nodeCollection))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+    /**
+     * Is checking if Number of InitialNodes is greater than one
+     *
+     * @param newNode
+     * @param nodeCollection
+     * @return true if number of initialNodes is more than one
+     */
+    private boolean isNumberOfInitialNodesIsToHigh(INode newNode, Collection<INode> nodeCollection)
+    {
+        if (newNode instanceof IInitialStateNode)
+        {
+            int numberOfInitialNodes = 0;
+
+            for (INode node : nodeCollection)
+            {
+                if (node instanceof IInitialStateNode)
+                {
+                    numberOfInitialNodes++;
+                }
+            }
+            if (numberOfInitialNodes > maxNumberOfInitialNodes)
+            {
+                return true;
```

Comment:
Ok it was my mistake. I though that you point the method checkIfNumberOfInitialNodesIsMoreThanOne not this one. I can provide changes and it will be look like this: 
 private boolean isNumberOfInitialNodesIsToHigh(INode newNode, Collection<INode> nodeCollection)
 {	
	boolean result =false;
	if (newNode instanceof IInitialStateNode)
 	{
             int numberOfInitialNodes = 0;
 
             for (INode node : nodeCollection)
             {
                 if (node instanceof IInitialStateNode)
                 {
                     numberOfInitialNodes++;
                 }
             }
	     result = numberOfInitialNodes > maxNumberOfInitialNodes;
         }
         return result;
 }
Can you tell me if it look's any better in your eyes now or you want to change it even more?